### PR TITLE
Bluetooth: services: Move GATT services source files

### PR DESCRIPTION
**Pull request by joerchan**
Move the GATT services source files. Components that are re-used are moved into the bluetooth subsystem and Kconfig options are added to include them in the build.
Services that are only used once are moved into the sample that includes it.
The IPSP service was not in use so it was removed.

### CODE REVIEW

**Comment by jhedberg** [comment]
In general I'm happy for the direction this is taking, however one thing to be aware of is that you're now introducing these as official public APIs. I think it might make sense to flag them as experimental for now, so we can spend some time seeing if they need any fine-tuning or not.

**Comment by joerchan** [comment]
@jhedberg Good point. Which is also why I didn't move all of them, only the ones that was being re-used. I have marked BAS and HRS as Experimental (Hopefully that is the correct way to do it, I wasn't really sure).

File: subsys/bluetooth/services/Kconfig.bas [hunk 1]
```diff
@@ -0,0 +1,29 @@
+# Kconfig - Bluetooth GATT Battery service
+#
+# Copyright (c) 2018 Nordic Semiconductor ASA
+#
+# SPDX-License-Identifier: Apache-2.0
+#
+
+menuconfig BT_GATT_BAS
+	bool "Enable GATT Battery service"
+	select SENSOR
```
Comment (lemrey) [inline subsys/bluetooth/services/Kconfig.bas]: This is no longer needed.

File: subsys/bluetooth/services/Kconfig.bas [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+	default n
+
+if BT_GATT_BAS
+
+config BT_GATT_BAS_LOG_LEVEL
```
Comment (lemrey) [inline subsys/bluetooth/services/Kconfig.bas]: We can use the Kconfig template for this now.

**Review by SebastianBoe — APPROVED** [review]

**Review by lemrey — COMMENTED** [review]
Perhaps then it'd be a good idea to give the user the option to not run `bas_init` during `SYS_INIT`, and make that API public?

**Comment by joerchan** [comment]
> Perhaps then it'd be a good idea to give the user the option to not run `bas_init` during `SYS_INIT`, and make that API public?

There is nothing happening in bas_init now. The service is registered as a static service.

**Comment by lemrey** [comment]
> There is nothing happening in bas_init now. The service is registered as a static service.

Right, so the question is whether we want to make the `bas_init` public now so that we don't have to introduce it afterwards if something gets added to it, and there could be a reason for users to run it manually (not run it during `SYS_INIT`).

**Comment by joerchan** [comment]
@lemrey I don't think that is necessary. And later if I'm wrong it should be no trouble to add it :)

**Review by lemrey — APPROVED** [review]

**Review by jhedberg — APPROVED** [review]

**Pull request merged**